### PR TITLE
task 6 GET /api/articles/:article_id/comments 1st draft complete

### PR DESCRIPTION
**Pull request by Bboymad**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -147,4 +147,58 @@ describe('GET /api/articles', () => {
...
+            'votes',
+            'created_at',
+            'author',
+            'body',
+            'article_id'
+            ];
+          body.comments.forEach((comment) => {
+            expectedProperties.forEach(property => {
+              expect(comment).toHaveProperty(property);
+          });
```
Comment (Harrison365) [inline __tests__/app.test.js:171]: Nice. We want to check the values too though. Including that the article_id is the one we asked for.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -147,4 +147,58 @@ describe('GET /api/articles', () => {
...
+      });
+    });
+    test('should return the comments sorted by date descending (most recent first)', () => {
+      return request(app)
+      .get('/api/articles/1/comments')
+      .then(({ body }) => {
+        expect(body.comments).toBeSortedBy('created_at', {
+          descending: true
+        })
+      });
```
Comment (Harrison365) [inline __tests__/app.test.js:182]: Nice 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -147,4 +147,58 @@ describe('GET /api/articles', () => {
...
+    test('should respond with 400 when invalid endpoint is used', () => {
+      return request(app)
+      .get('/api/articles/not_an_id/comments')
+      .expect(400)
+      .then(({ body }) => {
+      expect(body).toEqual({ msg: "Bad request" });
+      });
+    });
+  });
+});
```
Comment (Harrison365) [inline __tests__/app.test.js:203]: Nice sad path 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -147,4 +147,58 @@ describe('GET /api/articles', () => {
...
+
```
Comment (Harrison365) [inline __tests__/app.test.js:204]: Please add one more 200 test where you receive an empty array for an article that exists but has no comments.

File: models/model.articles.js [hunk 5]
```diff
@@ -38,4 +38,31 @@ exports.formatArticles = () => {
...
+      article_id = $1
+    ORDER BY
+      created_at DESC;`, [id])
+  .then(({ rows }) => {
+    const comments = rows
+    if (comments.length === 0) {
+      return Promise.reject({
+        status: 404,
+        msg: 'Article not found'
+      });
```
Comment (Harrison365) [inline models/model.articles.js:64]: Unfortunately, this won't work when we account for articles with no comments. You will need to do a separate check-articles-exist db.query.

**Comment by Harrison365** [comment]
Good work here 👍 
Just a few edits and additions required. 
Thanks :)
